src/Data/Vec/Properties.agda

```diff
@@ -860,12 +866,39 @@ foldr-∷ʳ : ∀ (B : ℕ → Set b) (f : FoldrOp A B) {e} y (ys : Vec A n) →
 foldr-∷ʳ B f y []       = refl
 foldr-∷ʳ B f y (x ∷ xs) = cong (f x) (foldr-∷ʳ B f y xs)
 
+-- init, last and _∷ʳ_
+
+init-∷ʳ : ∀ x (xs : Vec A n) → init (xs ∷ʳ x) ≡ xs
+init-∷ʳ x xs with xs ∷ʳ x in eq
```

Comment:
Are you sure this can't be done by induction on `xs` ?

Reply:
`init` is itself defined using `initLast`, so I think somehow I need to get `initLast (xs ∷ʳ x)` computing. I am not sure what to do with this goal:

```
init-∷ʳ x (y ∷ xs) = {!!}

Goal: (init (y ∷ (xs ∷ʳ x))
       | (initLast (y ∷ (xs ∷ʳ x)) | initLast (xs ∷ʳ x)))
      ≡ y ∷ xs
```

https://github.com/agda/agda-stdlib/blob/bfa4f8bc41da10bffaf69f5786301e587ae69c83/src/Data/Vec/Base.agda#L342-L349

Reply:
`init` should be defined as `take n xs`. You'll still need a `with` because of `splitAt`.  

These combinators that return proofs along with other things feel oddly out of place in `Data.Vec.Base`. I expected things here to all be pure computations.

Reply:
Do you suggest that I change the existing definition of `init` in `Data.Vec.Base`? Could that break existing code in the wild?

Reply:
If the definition of `init` and `last` were to change, how about I directly write an inductive definition?

Reply:
Yes, that is what I am suggesting. And "code" should not break, but proofs that rely on the exact definition might. But I doubt many people have such proofs.

Yes, a direct inductive definition would be better!

Reply:
Done. I like the straightforward definition better too. Thank you!